From 78c495c2667b916274a6569cce79458545372457 Mon Sep 17 00:00:00 2001
From: Philippe Normand <philn@igalia.com>
Date: Mon, 20 Jun 2016 11:45:53 +0200
Subject: [PATCH] [GStreamer][EME] don't wait forever on key negotiation

Error out after 5 seconds. Also added the predicate functor so that the
wait can be interrupted as soon as the key was received.
---
 .../graphics/gstreamer/WebKitCommonEncryptionDecryptorGStreamer.cpp     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WebCore/platform/graphics/gstreamer/WebKitCommonEncryptionDecryptorGStreamer.cpp b/Source/WebCore/platform/graphics/gstreamer/WebKitCommonEncryptionDecryptorGStreamer.cpp
index 4f7bb9d..11489748 100644
--- a/Source/WebCore/platform/graphics/gstreamer/WebKitCommonEncryptionDecryptorGStreamer.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/WebKitCommonEncryptionDecryptorGStreamer.cpp
@@ -210,7 +210,7 @@ static GstFlowReturn webkitMediaCommonEncryptionDecryptTransformInPlace(GstBaseT
             GST_DEBUG_OBJECT(self, "can't process key requests in less than PAUSED state");
             return GST_FLOW_NOT_SUPPORTED;
         }
-        priv->condition.wait(priv->mutex);
+        priv->condition.waitFor(priv->mutex, std::chrono::seconds(5), [priv] { return priv->keyReceived; });
         if (!priv->keyReceived) {
             GST_ERROR_OBJECT(self, "key not available");
             return GST_FLOW_NOT_SUPPORTED;
